src/c++/library/http_client.cc

```diff
@@ -196,6 +196,73 @@ CompressData(
   return Error::Success;
 }
 
+Error
+ParseSslCertType(
```

Comment:
ParseSslCertType and ParseSslKeyType are identical. Will there be a case where we support a specific Cert type but not that key type? If not, maybe we should re-use these functions

Reply:
There is an additional Key type that I have not added support for yet. It's called "ENG", where the key is generated from a crypto engine. We might need to add it in future. Hence keeping them separate.